src/components/ContactDetails/ContactDetailsAvatar.vue

```diff
@@ -133,6 +138,27 @@ export default {
 			}
 			return false
 		},
+		supportedNetworks() {
+			return ['facebook'] // TODO: get from api
+		},
+		hasSocialId() {
+			const jCal = this.contact.jCal
+			const socialId = jCal[1].filter(props => props[0] === 'x-socialprofile')
```

Comment:
Please use `this.contact.vCard.getAllProperties('x-socialprofile')`
This is the appropriate way of getting properties :)

`this.contact.vCard.getAllProperties('x-socialprofile').filter(prop => prop.getParameter('type').toLowerCase() === 'facebook')`

Reply:
Wow, 15 lines shrinked to one - very nice!
Added another `.toString()` to prevent TypeErrors when changed.